Update ImagePreviewFragment.kt

brownowski opened the pull request:
Fix banding in gradients by setting the preferred Bitmap config to Bitmap.Config.ARGB_8888 for SubsamplingScaleImageView

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit


- **Bug Fixes**
	- Improved image display quality by setting the preferred bitmap configuration to ARGB_8888.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

brownowski committed:
Update ImagePreviewFragment.kt

Set the preferred config to Bitmap.Config.ARGB_8888

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/cryptomator/android?pullRequest=526) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/cryptomator/android?pullRequest=526) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/cryptomator/android?pullRequest=526) it.</sub>

brownowski commented on presentation/src/main/java/org/cryptomator/presentation/ui/fragment/ImagePreviewFragment.kt:
```diff
@@ -65,6 +65,7 @@ class ImagePreviewFragment : Fragment() {
 			imagePreviewFile.uri?.let {
 				hideProgressBar()
 				imageView.orientation = SubsamplingScaleImageView.ORIENTATION_USE_EXIF
+				SubsamplingScaleImageView.setPreferredBitmapConfig(Bitmap.Config.ARGB_8888)
```
The method is a static method and has been tested.

SailReal reviewed (commented):
Thanks for your contribution! :green_heart:

Do you know if this is still a problem? https://github.com/davemorrissey/subsampling-scale-image-view/issues/366

brownowski commented:
> Thanks for your contribution! 💚
> 
> Do you know if this is still a problem? [davemorrissey/subsampling-scale-image-view#366](https://github.com/davemorrissey/subsampling-scale-image-view/issues/366)

Hmm, it might be.

I've run some tests, and I found that Bitmap.Config.HARDWARE performed better on my tablet. I had a hard time noticing any performance issues with ARGB_8888 on my phone, and HARDWARE seemed to work well too.

On my tablet I did notice some stuttering when scrolling between larger images with ARGB_8888, or panning on a few select files. This went away with HARDWARE, and it still also resolved the banding issue.

It might be worth testing it out on more devices, but switching to Bitmap.Config.HARDWARE might be better than ARGB_8888.

brownowski committed:
Update ImagePreviewFragment.kt

Changed from ARGB_8888 to HARDWARE for performance improvements. Quality remains the same as ARGB_8888.

brownowski committed:
Update ImagePreviewFragment.kt

Include missed import for android.graphics.Bitmap

SailReal commented:
Just to keep you up to date: I have done some more research and think this change needs more testing, but `Bitmap.Config.HARDWARE` sounds like the way to go to me.

SailReal commented:
This will also fix https://github.com/cryptomator/android/issues/482
